modules/openapi-generator/src/main/java/org/openapitools/codegen/languages/ScalaAkkaHttpServerCodegen.java

```diff
@@ -0,0 +1,486 @@
+        supportingFiles.add(new SupportingFile("multipartDirectives.mustache",
+                (sourceFolder + File.separator + invokerPackage).replace(".", java.io.File.separator), "MultipartDirectives.scala"));
+    }
+
+    private static final String IS_10_1_10_PLUS = "akkaHttp10_1_10_plus";
+    private boolean is10_1_10AndAbove = false;
+
+    private static final Pattern akkaVersionPattern = Pattern.compile("([0-9]+)(\\.([0-9]+))?(\\.([0-9]+))?(.\\+)?");
+    private void parseAkkaHttpVersion() {
+        Matcher matcher = akkaVersionPattern.matcher(akkaHttpVersion);
+        if (matcher.matches()) {
+            String majorS = matcher.group(1);
+            String minorS = matcher.group(3);
+            String patchS = matcher.group(5);
+            boolean andAbove = matcher.group(6) != null;
+            int major = -1, minor = -1, patch = -1;
+            try {
+                if (majorS != null) {
+                    major = Integer.parseInt(majorS);
+                    if (minorS != null) {
+                        minor = Integer.parseInt(minorS);
+                        if (patchS != null) {
+                            patch = Integer.parseInt(patchS);
+                        }
+                    }
+                }
+
+
+                if (major > 10 || major == -1 && andAbove) {
+                    is10_1_10AndAbove = true;
+                } else if (major == 10) {
+                    if (minor > 1 || minor == -1 && andAbove) {
+                        is10_1_10AndAbove = true;
+                    } else if (minor == 1) {
+                        if (patch >= 10 || patch == -1 && andAbove) {
+                            is10_1_10AndAbove = true;
+                        }
+                    }
+                }
+
+            } catch (NumberFormatException e) {
+                LOGGER.warn("Unable to parse " + AKKA_HTTP_VERSION + ": " + akkaHttpVersion + ", fallback to " + DEFAULT_AKKA_HTTP_VERSION);
+                akkaHttpVersion = DEFAULT_AKKA_HTTP_VERSION;
+                is10_1_10AndAbove = true;
+            }
+        }
+
+        additionalProperties.put(IS_10_1_10_PLUS, is10_1_10AndAbove);
+    }
+
+    @Override
+    public CodegenOperation fromOperation(String path, String httpMethod, Operation operation, List<Server> servers) {
+        CodegenOperation codegenOperation = super.fromOperation(path, httpMethod, operation, servers);
+        addPathMatcher(codegenOperation);
+        return codegenOperation;
+    }
+
+    @Override
+    public CodegenParameter fromParameter(Parameter parameter, Set<String> imports) {
+        CodegenParameter param = super.fromParameter(parameter, imports);
+        // Removing unhandled types
+        if(!primitiveParamTypes.contains(param.dataType)) {
+            param.dataType = "String";
+        }
+        if (!param.required) {
+            param.vendorExtensions.put("hasDefaultValue", param.defaultValue != null);
```

Comment:
We've trying to normalize the naming of vendor extensions. I wonder if you can rename `hasDefaultValue` to `x-has-default-value` instead.